### ISSUE TITLE
Unintuitive / overloaded behaviour of `fig_inches` option

### ISSUE THREAD

**Issue by maxalbert** [ioam/holoviews#420]
I have found the behaviour of the option `fig_inches` to be fairly unintuitive . The output of `hv.help` states: _"May be set as an integer in which case it will be used to autocompute a size. Alternatively may be set with an explicit tuple or list, in which case it will be applied directly after being scaled by fig_size."_

The latter part makes a lot of sense, and is in line with what matplotlib does IIRC. However, even after a bunch of experiments I never understood how the first part works, and the docs don't really explain it in their current form.

According to @philippjfr (on Gitter), "thinking about a sizing and aspect system that works well across backends should become a major priority after 1.5".

Suggestions in the short term:

- Restrict the meaning of `fig_inches` to the second option (= 2-tuple of width, height).

- Either disable the first meaning, or create a separate option for it. (Unfortunately I can't suggest a good name because I don't actually know what it does. ;-) ) In the latter case its meaning should be explained in more detail in the help string.

In fact, I just realised that matplotlib calls this option `figsize` instead of `fig_inches`, but this clashes with Holoviews' meaning of the `figsize` option (which specifies the "size relative to the supplied overall `fig_inches` in percent). Maybe it would be good to rename the Holoviews options as follows?

- `fig_inches` --> `figsize`
- `figsize` --> `fig_scaling`

(Or something similar?)

**Comment by jbednar** [comment]
I have found the behavior of *all* the HoloViews figure sizing parameters confusing; I can't tell which one to use in various scenarios (%output, figsize, fig_inches, etc.), nor how to achieve the sizes I want.  It's very frustrating.

That said, changing the name of an option to match the name of another existing option seems like  a recipe for even more confusion!

**Comment by maxalbert** [comment]
I agree with both points, and I personally would typically try to avoid renaming things to retain backwards compatibility where possible.

That said, it sounds like the entire figure sizing system will need a considerable overhaul and re-design. In such a situation it may be best to bite the bullet and accept renamings if they are justified by increased user friendliness, consistency and discoverability. It is always a possibility to emit a warning which states the changes clearly in case a re-named option is being used, combined with an easy option to turn the warning off once it has been acknowledged by the user.

**Comment by jbednar** [comment]
If it comes to that, I'd rather just pick an entirely new set of names, so that we can consistently detect the old ones.
